Src/ILGPU/Runtime/AcceleratorObject.cs

```diff
@@ -105,8 +105,8 @@ partial class Accelerator
         /// The list of linked child objects.
         /// </summary>
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
-        private List<WeakReference<AcceleratorObject>> childObjects =
-            new List<WeakReference<AcceleratorObject>>();
+        protected List<WeakReference<AcceleratorObject>> ChildObjects
```

Comment:
We should not expose this private property to anyone I guess; maybe we should define a thread-safe function to iterate over all instances of a specific type and invoke a user-defined callback; something like this:
````c#
protected void ForEachChildObject<T>(Action<T> callback) where T : AcceleratorObject
{
    lock (syncRoot)
    {
         ...
    }
}
```